### ISSUE TITLE
[TSDoc> Search]: Proper name is not defined for 'Typesense' link under search dialog.

### ISSUE THREAD

**Issue by vaishali1397** [microsoft/rushstack-websites#172]
**Test Environment:**
OS Version: 22H2 (OS Build 25370.1)
Browser: Edge Dev (Version 115.0.1866.1 (Official build) dev (64-bit))
URL: https://tsdoc.org/
Screen reader: Narrator

**Repro Steps:**

1. Open the above URL in edge dev browser and login with v-.
2. Turn on the narrator through Ctrl+ windows+ Enter key.
3. Navigate to the 'Search' button and activate it.
4. Navigate to the 'Search by: Typesense' link through tab key under search dialog.
5. Observe the issue.

**Actual Result:**
Name is not defined for the 'Typesense' link.
Screen reader is only announcing as **'Content information footer link search by'.**

**Expected Result:**
Proper name should be announced for 'Typesense' link.
Screen reader should announce as **'Content information footer link search by Typesense'.**

**Notes:**
Same issue repro on **Playground** that screen reader is not announcing any information of 'to press esc key to move out from the code sample section'.

**User Impact:**
Screen reader user will get impacted as user will not get the exact information conveyed to the user and hence will have difficulty in interacting with the content if purpose is not clearly announced.

**WCAG Reference Link:**
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

Have feedback to share on Bugs? Please help fill [Trusted Tester Bug Feedback (office.com)](https://nam06.safelinks.protection.outlook.com/?url=https://forms.office.com/pages/responsepage.aspx?id%3Dv4j5cvGGr0GRqy180BHbR5dUqGQgJ9lItv-jqY8F09lURVdYTENIRkY2UzlCR0NKV1E2VUE5OVpDVy4u&data=04%7c01%7cv-chisha%40microsoft.com%7c1334b110f98744a7f5df08d968eb842b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637656183104980596%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c1000&sdata=r4ndSHk/LrBDwzi7FO%2BjKN0DCnqUSUcELBvGRkgc5CU%3D&reserved=0)"

<img width="960" alt="Screen reader is not announcing descriptive information for the search by link" src="https://github.com/microsoft/rushstack-websites/assets/104483005/bffd970d-11d0-4374-857f-98fb028a2904">

https://github.com/microsoft/rushstack-websites/assets/104483005/f234f1d1-f034-421b-aa75-4df3309b2832
